### PR TITLE
feat: Add `StreamMode` to support blocks and records

### LINKED ISSUE #14704: Open PR with `STREAM_MODE` in handleWorkflow with default set to `RECORDS`

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #14704 

Adds a StreamMode that can help choose between producing blocks, records or both. Currently uses `Records` as the default.

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/BootstrapConfigProviderImpl.java [hunk 1]
```diff
@@ -57,11 +59,13 @@ public BootstrapConfigProviderImpl() {
                 .withSource(SystemEnvironmentConfigSource.getInstance())
                 .withSource(SystemPropertiesConfigSource.getInstance())
                 .withSource(new PropertyConfigSource(SEMANTIC_VERSION_PROPERTIES_DEFAULT_PATH, 500))
+                .withConfigDataType(BlockStreamConfig.class)
                 .withConfigDataType(BootstrapConfig.class)
                 .withConfigDataType(HederaConfig.class)
                 .withConfigDataType(FilesConfig.class)
                 .withConfigDataType(VersionConfig.class)
                 .withConfigDataType(LedgerConfig.class)
+                .withConfigDataType(AccountsConfig.class)
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/BootstrapConfigProviderImpl.java:68]: Intentional? I don't see anywhere else in this PR that this config is used

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/BootstrapConfigProviderImpl.java:68]: Just took changes from spike branch. I think that config was missing also in the bootstrap config and a unit test was failing.

**Review by mhess-swl — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @Neeharika-Sompalli !

**Review by rbarkerSL — APPROVED** [review]
LGTM

**Pull request merged**